mmf/models/vilbert.py

```diff
@@ -1155,6 +1159,24 @@ def forward(
         output_all_attention_masks=False,
     ):
 
+        batch_size = image_feature.size(0)
+        num_options = input_ids.size(1)
+
+        if self.config.input_preprocessor == 'retrieval':
+            image_feature = \
+                image_feature.view(-1,
+                                   image_feature.size(2),
+                                   image_feature.size(3)
+                                   )
+            image_location = \
+                image_location.view(-1,
+                                    image_location.size(2),
+                                    image_location.size(3)
+                                    )
+            input_ids = input_ids.view(-1, input_ids.size(2))
+            attention_mask = attention_mask.view(-1, attention_mask.size(2))
+            token_type_ids = token_type_ids.view(-1, token_type_ids.size(2))
+
```

Comment:
Its tricky. Then accessing batch_size and num_options becomes a problem. I had it before in get_image_and_text_features and then had to move it here because of that problem. Let me see if I can access these two values in any other way inside this function.